src/contracts/fund/trading/calls/isKyberTakePermitted.ts

```diff
@@ -0,0 +1,69 @@
+import web3EthAbi from 'web3-eth-abi';
+import { getContract } from '~/utils/solidity/getContract';
+import { Contracts, Exchanges } from '~/Contracts';
+import { getRoutes } from '~/contracts/fund/hub/calls/getRoutes';
+import { getHub } from '~/contracts/fund/hub/calls/getHub';
+import { FunctionSignatures } from '../utils/FunctionSignatures';
+import { Environment } from '~/utils/environment/Environment';
+import { Address, QuantityInterface } from '@melonproject/token-math';
+
+const isKyberTakePermitted = async (
```

Comment:
I'm not 100% sure, but this looks to me like a "duplicated code" smell. What is the actual difference between `isKyberTakePermitted` and `isOasisDexMakePermitted`. I only see fillTakerQuantity which could be defaulted to `0`.

So better to refactor this function to `isTakePermitted` that works for all exchanges. Like the contracts are implemented if I understand them right.

https://github.com/melonproject/protocol/blob/develop/src/contracts/fund/policies/risk-management/PriceTolerance.sol#L73:L79

Reply:
https://github.com/melonproject/protocol/blob/develop/src/contracts/fund/trading/calls/isOasisDexTakePermitted.ts#L47

Reply:
Right. Please use the Exchange enum here: https://github.com/melonproject/protocol/blob/develop/src/Contracts.ts#L269